### ISSUE TITLE
Speech I/O testing with TTS and ASR feedback loop

### ISSUE THREAD

**Issue by psibre** [EllaVator/EllaVator#16]

**Comment by psibre** [comment]
An unfinished implementation is in the `speech-test` branch. We still need to tie sphinx into the `recognize` method.

**Comment by naszka** [comment]
I'll try to do that today

**Comment by psibre** [comment]
@naszka cool, thanks! Chat with me if you run into low-level problems.

**Comment by psibre** [comment]
After hacking around a bit, I think we can do a similar thing to what the `SphinxASR` class does inside the [OpenDial Sphinx ASR plugin](https://github.com/plison/opendial/releases/download/version1.3b/sphinx-plugin.zip), which is to say,
```java
		configuration = new Configuration();
		configuration.setAcousticModelPath(ACOUSTIC_MODEL);
		configuration.setDictionaryPath(DICTIONARY);
		configuration.setGrammarPath(grammarFile.getParent() + "/");
		configuration.setGrammarName(grammarFile.getName().replace(".gram", ""));
		configuration.setUseGrammar(true);
		System.getProperties().setProperty("logLevel", "OFF");
		try {
			asr = new StreamSpeechRecognizer(configuration);
		}
		catch (IOException e) {
			throw new RuntimeException("cannot start Sphinx recognizer: " + e);
		}
```
followed by
```java
				asr.startRecognition(stream, new TimeFrame(15000));
				SpeechResult curResult = asr.getResult();
				log.info((curResult==null)? "No recognition results " : "Recognition completed");
				if (curResult != null) {
					Map<String, Double> results = createNBestList(curResult);
					system.addUserInput(results);
				}
				asr.stopRecognition();
```
adapted to our scenario of course.

**Comment by psibre** [comment]
Incidentally, after listening to some of the synthesized audio, I think even if sphinx works, it will fail to understand anything at all... =)

**Comment by naszka** [comment]
:) we could write a separate demo, using LiveSpeechRecognizer just to test our grammar. I have class until 2 but can play around with it after that.

**Comment by psibre** [comment]
@naszka I agree, we should hook some sphinx into our app, but that's a separate task I think.

**Comment by psibre** [comment]
See #24

**Comment by benti** [comment]
I tried to implement the test without MaryTTS using prepared audio files instead. You can find a working test in my repo in the branch `speech-test`: https://github.com/benti/EllaVator/tree/speech-test

Unfortunately, the recognition is not very good, but at least it recognizes the example I provided there.

**Comment by psibre** [comment]
Closed by #35
